packages/wasi/src/index.ts

```diff
@@ -783,6 +933,41 @@ export default class WASIDefault {
         (fd: number, iovs: number, iovsLen: number, nread: number) => {
           const stats = CHECK_FD(fd, WASI_RIGHT_FD_READ);
           const IS_STDIN = stats.real === 0;
+          if (this.asyncified) {
+            return asyncify(async () => {
+              let read = 0;
+              outer: for (const iov of getiovs(iovs, iovsLen)) {
+                let r = 0;
+                while (r < iov.byteLength) {
+                  let length = iov.byteLength - r;
+                  let position =
+                    IS_STDIN || stats.offset === undefined
+                      ? null
+                      : Number(stats.offset);
+                  let rr: number = await promisify(fs.read)(
```

Comment:
I think it would be better to use the `readSync` API rather than make it awaitable and wait for the result.

```suggestion
                  let rr: number = fs.readSync(
```